### DISCUSSION TITLE
PCB trouble shooting help

### DISCUSSION THREAD

**Discussion by rjbutler** [doudar/SmartSpin2k#254] [Q&A] [answered]
I flashed the ESP32 and set it up, built the PCB and tested everything out.  It all was working on the bench so I closed up the case, put it on the bike and plugged it all in.  I hit the shifters and it turned the resistance knob.  Great!  I did it a couple more times, and it stopped.  The blue LED went out and the red LED was on solid.  I can no longer connect to the ESP32, even over USB with the PCB removed from the case.  

Any suggestions before I try to de-solder it and remove the ESP32?  I thought maybe the twisting had caused a short, but everything still looks good.  Solder connections look good too.  

I'd start over, but it's hard to find all the parts.  Is there an alternative to the TMC2225 (2208?  2226?  I don't know enough to know if they are drop in replacements).  And the RECOM part is also hard to find.

Thanks

**Comment by doudar** [comment, marked as answer]
I believe a 2208 step stick should be a drop in replacement but I haven’t personally tried it. 

I wonder if it’s a possible infant mortality issue on the esp32? Normally they either work or they don’t. I’ve never experienced one that worked for a short time only. 

Make sure that none of the legs from the tmc are touching the back side of the esp32 and that your power supply is less than 24v dc. 

Try to re-flash the esp32 as well. Maybe the entire memory was corrupted somehow. 

The only other thing I can think of is to make sure the wires didn’t get pinched when you closed the case and that the stepper motor wire is completely attached and good. I’ve had the TMC driver lockup an esp32 a couple of times and removing the stepper motor cable fixed the issue. If it works with the stepper motor cable disconnected, make a new cable and try again. 

And maybe triple check your power supply for voltage and amperage. I’ve had some of the cheap Chinese supplies with voltages way outside of spec and amperage much lower than labeled. 

Have fun and keep us all up to date!

**Reply by doudar** [reply]
Also, you could use a lm7805 as a drop in replacement for the recom but they run very hot in this application and so I don’t like them being in a closed box. Maybe it’s okay if you attached a small heat sink to it but please monitor it closely if you do. 

If you use the lm7805, make sure your input power supply is no more than 12v-14v

**Reply by rjbutler** [reply]
Unfortunately neither the flasher app nor PlatformIO will connect to the esp32.  Thanks for the suggestions on alternates.  I'll try to remove the esp32 and test some voltages on the rest of the board.  I have a 24v 2.2A power supply, it's a salvage from an old Kodak photo printer, and I'm pretty sure it's good but I'll double check the voltage just in case.  Thanks

**Reply by rjbutler** [reply]
@doudar I just want to check something.  On the PCB video, it really looks like the 10uf capacitor is being placed on the right side of the board under the 2 pin input connector, but the silk screen says 100uf.  I used 100uf and it was a tight fit, so I'm wondering if the board is labeled wrong?  I looked around the repo for a schematic, but didn't find one.

**Reply by doudar** [reply]
As far as capacitor values, I’ve found it should have an input capacitor to filter ripples caused by the stepper driver but if you have a good power supply (and it sounds like you probably have an excellent power supply) it will work without even having one as the recom is very robust. 

So, in general, anything 3.3uf-100uf is fine and shouldn’t cause issues. The two capacitors (especially the ceramic capacitor) on the output side are more important.

**Reply by rjbutler** [reply]
@doudar Sorry if I’m being dense, on the video (you?) say the 50v cap on the input side is important.  The BOM on the pcbway site lists:
10uf 50v capacitor  x 1 
100uf 25v capacitor x 1

The board looks like it’s labeled as 100uf on the input side, so I used a 100uf 50v.   And the board on the output side is labeled as 10uf (I also used a 50v here).   Is the board labeling reversed?  Or is the BOM wrong?

![35C3944B-92BB-40B3-9C6A-615AF28A3D42](https://user-images.githubusercontent.com/505686/143468580-e7662275-09d4-45ca-9827-350e60c9abc1.jpeg)

**Reply by doudar** [reply]
No problem! 

Sorry if it's not clear. It's important that the capacitor on the input side needs be rated for 50v (or double your power supply voltage). The capacitance value (uf) on the input is not as important but in general, bigger is better as it compensates for when the stepper is drawing large pulses of power and 100uf is the absolute maximum value that you need in that location.

**Reply by rjbutler** [reply]
OK, so on the output side is 10uf enough?

**Reply by doudar** [reply]
Yes

**Reply by doudar** [reply]
Thinking back, I have had a TMC driver fail which prevented the ESP32 from booting. It will be interesting to hear what you determine happened.

**Reply by rjbutler** [reply]
I ordered some TMC2208’s.  Maybe I’ll be able to salvage the recom.

**Reply by rjbutler** [reply]
I think you may be right, @doudar.  I pulled the ESP32 and it does boot OK off the board.  I'm getting 5v on the RECOM, so I assume that's OK so probably the TMC2225?  I did test the power supply and it's outputting 25v.  I assume that's OK?  Or should I buy a new one?  Thanks for the help.

**Reply by doudar** [reply]
Both the recom and the tmc2225 (I didn’t look up the 2208) are good to over 30v.
